dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java

```diff
@@ -504,13 +496,20 @@ private void revertLoginAsSessionInfo(final HttpServletRequest request, final Ho
 	 * @param req    The {@link HttpServletRequest} object.
 	 * @param userID The ID of the user that is being impersonated.
 	 *
-	 * @throws DotDataException     An error ocurred when accessing the data source.
+	 * @throws DotDataException     An error occurred when accessing the data source.
 	 * @throws DotSecurityException The specified user does not have permissions to perform this action.
+	 * @throws DotRuntimeException  There are missing session attributes, or the User doesn't have access to any Site.
 	 */
 	private void setImpersonatedUserSite(final HttpServletRequest req, final String userID) throws DotDataException, DotSecurityException {
 		final HttpSession session = req.getSession();
 		final User user = this.userAPI.loadUserById(userID);
 		final String currentSiteID = (String) session.getAttribute(com.dotmarketing.util.WebKeys.CMS_SELECTED_HOST_ID);
+		if (UtilMethods.isNotSet(currentSiteID)) {
+			final String errorMsg = String.format("The CMS_SELECTED_HOST_ID attribute is not present in the session " +
+					"for User '%s'", userID);
+			Logger.error(this, errorMsg);
+			throw new DotRuntimeException(errorMsg);
```

Comment:
Should we throw here or should we get the current host based on the hostWebApi?

Reply:
As per discussion in Lunik's channel, we're throwing an error because this is a situation that must never happen, and was actually the first symptom that indicated failures in the Redis-based Tomcat Session plugin. This is the reason why we're not recovering from this issue.